### DISCUSSION TITLE
Can't see my code changes in plausible self hosting server

### DISCUSSION THREAD

**Discussion by Lalarukh-khan** [plausible/analytics#2609] [Self-Hosted Support] [answered]
Hey There!
I've implemented the self-hosted plausible project locally in my mac. Its currently running on localhost:8000. 
But when I made changes in frontend code. For example in header I tried to change the "Repo" option to the "Plausible Git". 
But when I reloaded in browser it didn't showed my frontend changes. 
For testing purpose I removed the "layout" folder from the project but still the frontend was working. However the frontend layout folder was deleted. 
Let me know If i can make changes in the plausible self-hosted code and see its changes. 
Thanks

**Comment by ruslandoga** [comment, marked as answer]
👋 @Lalarukh-khan

Please see [`CONTRIBUTING.md`](https://github.com/plausible/analytics/blob/master/CONTRIBUTING.md) for local development setup instructions.

**Reply by ruslandoga** [reply]
https://hexdocs.pm/phoenix/up_and_running.html might also be useful. Phoenix is the web framework that Plausible uses.

**Reply by Lalarukh-khan** [reply]
@ruslandoga I followed the above url of Contributing.md information. I'm an elixir developer too. My plausible project has already been deployed in elixir phoenix. I'm running also my plausible server on http://localhost:8000. When I run the command elixir command: 
mix ecto.create 
My plausible_db is already deployed in postgres but clickhouse DB is araising bugs, however its running perfectly in docker and plausible server. 
`mix ecto.create` returned me this:
`15:18:15.125 [error] GenServer #PID<0.585.0> terminating
** (UndefinedFunctionError) function :econnrefused.exception/1 is undefined (module :econnrefused is not available)
    :econnrefused.exception([])
    (db_connection 2.4.2) lib/db_connection/connection.ex:100: DBConnection.Connection.connect/2
    (connection 1.1.0) lib/connection.ex:622: Connection.enter_connect/5
    (stdlib 4.2) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
Last message: nil
State: Clickhousex.Protocol`
**`** (Mix) The database for Plausible.ClickhouseRepo couldn't be created: killed`**

**Reply by ruslandoga** [reply]
How do you run ClickHouse? What're the exact commands you use and the exact output you get when running locally, setting everything up?

**Reply by Lalarukh-khan** [reply]
I'm running `docker-compose up -d` to run plausible in docker. 
This to run the server in 8000 port: `docker logs plainstats-adam-wip-plausible-1 --follow`

**Reply by ruslandoga** [reply]
Note that when you run docker compose setup from `plausible/hosting` repo, no changes to the source code in `plausible/analytics` would affect it until you rebuild the container. Also note that this setup is not what [`CONTRIBUTING.md`](https://github.com/plausible/analytics/blob/master/CONTRIBUTING.md) recommends for development.

**Reply by Lalarukh-khan** [reply]
Is there any way to run it locally, any way to make changes in server and see them locally? 
Seeking your response.

**Reply by ruslandoga** [reply]
Yes, please see [`CONTRIBUTING.md`](https://github.com/plausible/analytics/blob/master/CONTRIBUTING.md).
